gazelle/pythonconfig/pythonconfig.go

```diff
@@ -126,7 +126,7 @@ type Configs map[string]*Config
 
 // ParentForPackage returns the parent Config for the given Bazel package.
 func (c *Configs) ParentForPackage(pkg string) *Config {
-	dir := filepath.Dir(pkg)
+	dir := path.Dir(pkg)
```

Comment:
Looking at the code, I am not sure it is working the way you think. `path.Dir` will just split at `/` if I understand things correctly and `filepath.Path` is much more sophisticated and I am surprised that this is not working correctly.

I don't have a Windows machine, but the code change looks suspicious.

Reply:
@aignas the same issue I have fixed [here](https://github.com/bazel-contrib/rules_jvm/pull/261). My initial approach was to use `cfgs[filepath.FromSlash(rel)]` but a more targeted fix was prosed in [this](https://github.com/bazel-contrib/rules_jvm/pull/261#issuecomment-2091026233) comment.

Basically, it fails because `pythonconfig.go#ParentForPackage` delegates to filepath.Dir(pkg) to replace the separator character "\" with "/", hence for a 2-levels deep path, e.g. rootDir\nestedDir1\nestedDir2, a parent is searched with "/" (rootDir/nestedDir1) but Configs contains a key with "\" (rootDir\nestedDir1).

Reply:
Ah, I see, thanks for the context, it was much needed here. Could you please rebase and add Changelog notes to the Changelog.md please?